.github/workflows/lint.yml

```diff
@@ -153,3 +153,20 @@ jobs:
           #   - Waiting for https://github.com/mackwic/colored/pull/122 to be merged.
 
           args: --file Cargo.lock --deny warnings --ignore RUSTSEC-2021-0145
+
+  no_std:
+    name: Check `no_std` compatability
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the Source Code
+        uses: actions/checkout@v3
+
+      - name: Set up Rust
+        uses: ./.github/actions/setup-rust
+
+      - name: Check
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --no-default-features -F serde -p iota-sdk
```

Comment:
How is this checking for no_wtd compatibility? Missing a target?

Reply:
It's just the default features minus `std`.

Reply:
Yeah but we need to compile it for a no_std target otherwise we may have std dependencies and we wouldn't know about it

Reply:
Not really true, since `#[no_std]` is set by omitting the `std` feature. However, this IS slightly more error prone as changing the `serde` feature to include `std` would break it.

Reply:
In an ideal world yes, but in the end `#[no_std]` is not enough. We had the same issue with bee and we opened this issue https://github.com/rust-lang/cargo/issues/8798. You can actually compile something with no_std that wouldn't compile on an actual no_std platform.

Reply:
How annoying. I'll add the target then.